File: code_blocks/tools/paywalls_custom_variables_1.kt
import androidx.compose.runtime.Composable
import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.ui.revenuecatui.PaywallDialog
import com.revenuecat.purchases.ui.revenuecatui.PaywallOptions
import com.revenuecat.purchases.ui.revenuecatui.CustomVariableValue

@Composable
fun PaywallWithCustomVariables(
    customerInfo: CustomerInfo,
    onDismiss: () -> Unit
) {
    val options = PaywallOptions.Builder { /* dismiss */ }
        .setCustomVariables(
            mapOf(
                "player_name" to CustomVariableValue.String("John"),
                "level" to CustomVariableValue.String("42")
            )
        )
        .build()

    PaywallDialog(
        options = options,
        shouldDisplayBlock = { !customerInfo.entitlements.active.containsKey("pro") }
    )
}

File: code_blocks/tools/paywalls_custom_variables_1.dart
import 'dart:developer';

import 'package:purchases_ui_flutter/purchases_ui_flutter.dart';

void presentPaywallWithCustomVariables() async {
  final paywallResult = await RevenueCatUI.presentPaywall(
    customVariables: {
      'player_name': CustomVariableValue.string('John'),
      'level': CustomVariableValue.string('42'),
    },
  );
  log('Paywall result: $paywallResult');
}
